File: 377-CombinationSumIv/377-CombinationSumIv.py
# Last updated: 12/6/2025, 5:50:04 am
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        nums.sort()
        dp = [0] * (target+1) 
        dp[0] = 1

        for i in range(target):
            for num in nums:
                if num+i > target: break
                dp[num+i] += dp[i]
        
        return dp[target]

File: 377-CombinationSumIv/377-CombinationSumIv.cpp
// Last updated: 12/6/2025, 5:50:08 am
class Solution {
public:
    int combinationSum4(vector<int>& nums, int target) {
        vector<unsigned int> dp(target + 1, 0);
        dp[0] = 1;
        for (unsigned int i=0; i<=target; i++){
            for (int num : nums){
                if (i+num <= target) dp[i+num] += dp[i];
            }
        }
        return dp[target];
    }
};